packages/perseus-core/src/widgets/radio/radio-upgrade.ts

```diff
@@ -1,8 +1,31 @@
-import type {PerseusRadioWidgetOptions} from "../../data-schema";
+import type {
+    PerseusRadioChoice,
+    PerseusRadioWidgetOptions,
+} from "../../data-schema";
+
+export const currentVersion = {major: 2, minor: 0};
+
+export function deriveNumCorrect(options: PerseusRadioWidgetOptions) {
+    const {choices, numCorrect} = options;
 
-export const currentVersion = {major: 1, minor: 0};
+    return (
+        numCorrect ??
+        choices.reduce(
+            (acc: number, curr: PerseusRadioChoice) =>
+                curr.correct ? acc + 1 : acc,
+            0,
+        )
+    );
```

Comment:
I see this algorithm was copied from the component code. I think there's a simpler way. I suspect the JIT can optimize this:

```ts
choices.filter((c) => c.correct).length
```

to avoid creating the intermediate array. Even if it doesn't... 🤷